From 50a992d364e12039c1a6ef73fc1396d3b8e73b80 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@st.com>
Date: Fri, 9 Sep 2016 09:44:24 +0200
Subject: [PATCH 3/5] MALI400: mali: replace get_unused_fd() by
 get_unused_fd_flags(0)

This patch replaces call to get_unused_fd() with equivalent call to
get_unused_fd_flags(0) to preserve current behavior for existing code.
Necessary for Android build.

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@st.com>
---
 drivers/gpu/arm/utgard/linux/mali_sync.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/arm/utgard/linux/mali_sync.c b/drivers/gpu/arm/utgard/linux/mali_sync.c
index 68868ca..dc1e3a2 100755
--- a/drivers/gpu/arm/utgard/linux/mali_sync.c
+++ b/drivers/gpu/arm/utgard/linux/mali_sync.c
@@ -18,6 +18,7 @@
 #include <linux/file.h>
 #include <linux/seq_file.h>
 #include <linux/module.h>
+#include <linux/fcntl.h>
 
 struct mali_sync_pt {
 	struct sync_pt         sync_pt;
@@ -298,7 +299,7 @@ s32 mali_sync_fence_fd_alloc(struct sync_fence *sync_fence)
 {
 	s32 fd = -1;
 
-	fd = get_unused_fd();
+	fd = get_unused_fd_flags(O_CLOEXEC);
 	if (fd < 0) {
 		sync_fence_put(sync_fence);
 		return -1;
-- 
1.9.1

